src/app/InteractionModelEngine.h

```diff
@@ -365,6 +365,28 @@ CHIP_ERROR ReadSingleClusterData(const Access::SubjectDescriptor & aSubjectDescr
                                  const ConcreteReadAttributePath & aPath, AttributeReportIBs::Builder & aAttributeReports,
                                  AttributeValueEncoder::AttributeEncodeState * apEncoderState);
 
+/**
+ *  Checks the preconditions required before writing an attribute, including:
+ *   1) The attribute must exist in the attribute catagory and it is read only.
```

Comment:
I don't know what "the attribute catagory" is, but certainly an attribute being read only is not a precondition for _writing_ it.

Reply:
This function is not used, and should be removed.